doc/source/packaging/build-systems.rst

```diff
@@ -0,0 +1,173 @@
+.. _ref_build_system:
+
+############
+Build System
+############
+
+The build system is a fundamental tool for packaging Python
+libraries. It generates distribution files that can be shared with
+users and developers. 
+
+
+Artifacts
+=========
+
+The build-system allows maintainers to generate artifacts for their Python
+libraries. Here, `artifacts` refers to both wheel and source files:
+
+- ``Wheel files`` have the ``*.whl`` file extension.
+- ``Source files`` have the ``*.tar.gz`` or ``*.zip`` extension.
+
+These are the files to upload to `PyPI`_ when releasing a new version of a
+PyAnsys project.
+
+.. warning::
+
+   Not all files are included by default in the source distribution. A
+   `MANIFEST.in`_ is required at the root of the project to specify additional
+   files.
+
+
+The interaction between the maintainer and the build-system is performed using a
+build-system tool. This tool provides a frontend and a backend. The maintainers
+trigger the frontend, which then calls the backend. The backend then reads the
+project directory and generates the artifacts, as :numref:`build system diag`:
```

Comment:
```suggestion
build system tool. This tool provides both a frontend and a backend. The maintainers
trigger the frontend, which then calls the backend to read the
project directory and generate the artifacts, as :numref:`build system diag`:
```

Reply:
@jorgepiloto Should the ", as" at the end of the last sentence be ", as shown in this diagram." I'm not sure how this directive as is renders.

Reply:
Thanks for spotting this, @PipKat. The ":numref:`name of reference`" render as `Fig.X`, where `X` is the ith-figure of the docs. Therefore, this could be refactored as:

```suggestion
build-system tool. This tool provides a frontend and a backend. The maintainers
trigger the frontend, which then calls the backend. The backend then reads the
project directory and generates the artifacts, as shown in :numref:`build system diag`:
```